services/opsgenie/service.go

```diff
@@ -122,71 +133,111 @@ func (s *Service) Alert(teams []string, recipients []string, messageType, messag
 		b := bytes.NewReader(body)
 		dec := json.NewDecoder(b)
 		dec.Decode(r)
-		return errors.New(r.Message)
+		return fmt.Errorf("opsgenie error: %s", r.Message)
 	}
 	return nil
 }
 
-func (s *Service) preparePost(teams []string, recipients []string, messageType, message, entityID string, t time.Time, details models.Result) (string, io.Reader, error) {
+func (s *Service) preparePost(teams []string, recipients []string, level alert.Level, message, entityID string, t time.Time, details models.Result) (*http.Request, error) {
 	c := s.config()
 	if !c.Enabled {
-		return "", nil, errors.New("service is not enabled")
+		return nil, errors.New("service is not enabled")
 	}
 
 	ogData := make(map[string]interface{})
-	url := c.URL + "/"
-
-	ogData["apiKey"] = c.APIKey
-	ogData["entity"] = entityID
-	ogData["alias"] = entityID
-	ogData["message"] = message
-	ogData["note"] = ""
-	ogData["monitoring_tool"] = "kapacitor"
+	u := c.URL
+	switch level {
+	case alert.OK:
+		up, err := url.Parse(c.URL)
+		if err != nil {
+			return nil, err
+		}
+		up.Path = path.Join(up.Path, entityID, "close")
```

Comment:
does this mean that close will always be the behavior on alert.OK? I think in the current API you can set whether this closes the alert or does some other action.

Reply:
No, just a first pass implementation. See my note above about how to make the change be backwards compatible since the user can no longer just specify the url directly as the URL needs to be constructed dynamically.

Reply:
cool, thanks.